modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeMemberResolver.java

```diff
@@ -19,46 +19,39 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import io.scalecube.cluster.Member;
-import org.apache.ignite.network.NetworkMember;
+import org.apache.ignite.network.ClusterNode;
 
 import static java.util.Objects.requireNonNull;
 
 /**
  * Resolver for scalecube specific member.
  */
-public class ScaleCubeMemberResolver {
+final class ScaleCubeMemberResolver {
     /** Map of public network member by its unique name. */
-    private final Map<String, NetworkMember> directMemberMap = new ConcurrentHashMap<>();
+    private final Map<String, ClusterNode> directMemberMap = new ConcurrentHashMap<>();
```

Comment:
Looks like these maps are never cleared. Maybe they should on node leaving ?

Reply:
I didn't change this code, it was already present. But let's introduce a solution for this

Reply:
fixed